packages/kit/src/utils/fork.js

```diff
@@ -36,34 +34,31 @@ export function forked(module, callback) {
 	 */
 	const fn = function (opts) {
 		return new Promise((fulfil, reject) => {
-			const child = child_process.fork(fileURLToPath(module), {
-				stdio: 'inherit',
+			const worker = new Worker(fileURLToPath(module), {
 				env: {
-					...process.env,
-					SVELTEKIT_FORK: 'true'
-				},
-				serialization: 'advanced'
+					...process.env
+				}
 			});
 
-			child.on(
+			worker.on(
 				'message',
-				/** @param {any} data */ (data) => {
+				/** @param {any} data */ async (data) => {
```

Comment:
I think this can still be sync?

```suggestion
				/** @param {any} data */ (data) => {
```